profiles/base/deployment.yaml

```diff
@@ -13,6 +14,8 @@ spec:
         - $(userid-header)
         - "-userid-prefix"
         - $(userid-prefix)
+        - "-workload-identity"
```

Comment:
Should this be a GCP specific overlay? What does gcp-sa get set for in non GCP configs? The empty string?

Reply:
Actually won't we get a bug if the gcp-sa parameter isn't defined?
I think a GCP specific overlay might be better.

Reply:
Sure
if it's empty string, then the plugin would not be executed.

Reply:
I stand corrected. It looks like the way this works is that kfctl is editing
base/params.env
https://github.com/kubeflow/kubeflow/blob/fa8886eae006dee0270bcc8bc1f5efc9e3c82e48/bootstrap/pkg/kfapp/kustomize/kustomize.go#L645

The logic updates any parameters that are set there.
So if the default in params.env is gcp_sa="" then it should evaluate correctly.